### PR TITLE
feat: add hilla-gradle-plugin

### LINKED ISSUE #900: Add Engine Gradle Plugin
For the Gradle support in Hilla we need to add a Gradle plugin that:
- enables running Hilla parser and generator
- allows to configure Hilla parser and generator on a similar level with Engine Maven Plugin

### LINKED ISSUE #901: Enable running Engine Gradle Plugin tasks from GradleRunner
Currently, [GradleRunner](https://github.com/vaadin/hilla/blob/e4ea08f1bbd6bef774ad91fac83838dec4816529/packages/java/engine-core/src/main/java/dev/hilla/engine/commandrunner/GradleRunner.java) does not actually support running Gradle.

After #900 and vaadin/platform#4050 are done, let's enable running Gradle form GradleRunner to support the configure task from Engine Gradle Plugin.

### LINKED ISSUE #902: Add IT test module with Gradle
To verify that Gradle support works in Hilla, let us add an IT test module with Gradle to the Hilla repository. Make sure:

- it uses the Gradle Plugin;
- generates TypeScript endpoints;
- applies configuration for parser packages;
- a “hello world” endpoint works when running the application.

### PR DESCRIPTION
**Pull request by taefi**
## Description

This adds the hilla-gradle-plugin and addresses the following issues:

Fixes #900
Fixes #901
Fixes #902 
Fixes vaadin/platform#4050

## Type of change

- [ ] Bugfix
- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [x] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: packages/java/gradle-plugin/build.gradle [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+/***********************************************************************************************************************
+ *
+ * Deployment
+ *
+ ***********************************************************************************************************************/
+gradlePlugin {
+    testSourceSets sourceSets.functionalTest
+    plugins {
+        hillaPlugin {
+            id = 'dev.hilla.engine'
```
Comment (taefi) [inline packages/java/gradle-plugin/build.gradle:145]:
```suggestion
            id = 'dev.hilla'
```

File: packages/java/gradle-plugin/build.gradle [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    mavenLocal()
+    mavenCentral()
+    maven { url = 'https://plugins.gradle.org/m2/' }
+    maven { url = 'https://maven.vaadin.com/vaadin-prereleases' }
+}
+
+dependencies {
+    implementation('org.jetbrains.kotlin:kotlin-stdlib:1.7.20')
+    implementation("dev.hilla:engine-core:${pom.parent.version}")
+    implementation("com.vaadin:flow-gradle-plugin:$flowVersion")
```
Comment (taefi) [inline packages/java/gradle-plugin/build.gradle:67]: read `flowVersion` from the pom.xml hierarchy instead.

Comment (taefi) [reply packages/java/gradle-plugin/build.gradle:67]: Read from the root pom.xml:
```
def rootPom = new XmlSlurper().parse(rootProject.file('../../../../pom.xml'))
```

File: packages/java/gradle-plugin/src/main/kotlin/dev/hilla/gradle/plugin/EngineGenerateTask.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
+/**
+ *    Copyright 2000-2022 Vaadin Ltd
```
Comment (taefi) [inline packages/java/gradle-plugin/src/main/kotlin/dev/hilla/gradle/plugin/EngineGenerateTask.kt:2]:
```suggestion
 *    Copyright 2000-2023 Vaadin Ltd
```

File: packages/java/tests/gradle/single-module-tests/gradle.properties [hunk 4]
```diff
@@ -0,0 +1,2 @@
+org.gradle.daemon=false
+hillaVersion=2.1-SNAPSHOT
```
Comment (taefi) [inline packages/java/tests/gradle/single-module-tests/gradle.properties:2]: drop this and read the version from the pom.xml:
```suggestion
```

File: packages/java/tests/gradle/single-module-tests/package.json [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    "@hilla/form": "2.1.0-alpha4",
+    "@hilla/frontend": "2.1.0-alpha4",
+    "@hilla/generator-typescript-cli": "2.1.0-alpha4",
+    "@hilla/generator-typescript-core": "2.1.0-alpha4",
+    "@hilla/generator-typescript-plugin-backbone": "2.1.0-alpha4",
+    "@hilla/generator-typescript-plugin-barrel": "2.1.0-alpha4",
+    "@hilla/generator-typescript-plugin-client": "2.1.0-alpha4",
+    "@hilla/generator-typescript-plugin-model": "2.1.0-alpha4",
+    "@hilla/generator-typescript-plugin-push": "2.1.0-alpha4",
+    "@hilla/generator-typescript-utils": "2.1.0-alpha4",
```
Comment (taefi) [inline packages/java/tests/gradle/single-module-tests/package.json:14]: use the symlinks to the current project

**Review by cromoteca — APPROVED** [review]

**Pull request merged**